From abd5806ee21979ef69388f03aa51d001d3782ee5 Mon Sep 17 00:00:00 2001
From: Tomohiro Kashiwada <kikairoya@gmail.com>
Date: Mon, 9 Jun 2025 02:06:03 +0900
Subject: [PATCH] [LLD][Cygwin] Add libcygwin and libmsys-2.0 to
 exclude-from-auto-export library list (#143118)

Linking for Cygwin target always needs -lcygwin (and, -lmsys-2.0
instead for MSYS2 target) but should not auto-export from
them, same as -lmingw32 for MinGW target.
---
 lld/COFF/MinGW.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lld/COFF/MinGW.cpp b/lld/COFF/MinGW.cpp
index 097cf228f7..5de61a001a 100644
--- a/lld/COFF/MinGW.cpp
+++ b/lld/COFF/MinGW.cpp
@@ -48,6 +48,8 @@ AutoExporter::AutoExporter(
       "libclang_rt.profile-arm",
       "libclang_rt.profile-i386",
       "libclang_rt.profile-x86_64",
+      "libcygwin",
+      "libmsys-2.0",
       "libc++",
       "libc++abi",
       "libFortranRuntime",
-- 
2.50.1.windows.1

